### ISSUE TITLE
Feature request: PriorityQueue#clear

### ISSUE THREAD

**Issue by smilingthax** [janogonzalez/priorityqueuejs#8]
Maybe you want to provide an official way to clear the queue (in O(1))? 
This is what is do:

    PriorityQueue.prototype.clear = function() {
       this._elements = [];
    };

One might even optionally allow `clear(arrayOfNewElements)` by implementing the known Array->Heap `heapify` operation (and not by calling `enq` repeatedly...).
OTHO I've yet to see a need for that, and just the `clear()` is much simpler and more useful.

**Comment by janogonzalez** [comment]
I'm kinda reluctant to add new features, I think in this case one can simply create a new priority queue object instead of calling a method to clear it.

**Comment by xgbuils** [comment]
I don't understand the meaning to clear priority with method heapify.
a) "heapify" creates a heap based on array. ( O(n) complexity )
b) "clear" clears priority queue.
What is the relationship?

**Comment by smilingthax** [comment]
@xgbuils: `clear()` is the special case of calling heapify with an empty array (heapify can't preserve existing queue entries).

@janogonzalez: To construct a new priority queue object, I need to provide my custom comparator again. And before I resort to peeking at `pq._comparator` I can just as well set `pq._elements=[]`. So without a official `clear()` method, I either have to rely on a non-public field of the priority queue, or keep the comparator around myself. 
IMHO the suggested `clear()` method (forget about the heapify thing) is both a very lightweight and beneficial addition.
Otherwise I'll continue my monkey-patching approach, relying on `_elements` not going away(?).

**Comment by xgbuils** [comment]
It makes sense to me.

**Comment by xgbuils** [comment]
I created a `heapify` method. I just encapsulated PQ#deq() while-loop into `sink` function and I reused this in `heapify` method.
https://github.com/xgbuils/priorityqueuejs/commit/f5f6422171c33b0f4eca0ad80e38ce921c69a805
Advantages:
- enqueue n items is running in O(n) time, not O(nlog(n))
- queue.heapify([]) clears elements and not, comparator.

Take a look and if you agree I can create a pull request.
